tests/reference_images.rs

```diff
@@ -39,6 +39,72 @@ where
     }
 }
 
+#[cfg(feature = "tiff")]
+#[test]
+fn compress_deflate() {
+    use image::codecs::tiff::{CompressionType, TiffDeflateLevel, TiffEncoder};
+
+    process_images(IMAGE_DIR, Some("tiff"), |base, path, _| {
+        println!("compress_images {}", path.display());
+        let img = match image::open(&path) {
+            Ok(img) => img,
+            // Do not fail on unsupported error
+            // This might happen because the testsuite contains unsupported images
+            // or because a specific decoder included via a feature.
+            Err(image::ImageError::Unsupported(e)) => {
+                println!("UNSUPPORTED {}: {e}", path.display());
+                return;
+            }
+            Err(err) => panic!("decoding of {path:?} failed with: {err}"),
+        };
+
+        let mut out_path = base.clone();
+        out_path.push(OUTPUT_DIR);
+        out_path.push("compressed_tiff");
+        out_path.push("deflate");
+        std::fs::create_dir_all(&out_path).unwrap();
+        out_path.push(path.file_name().unwrap());
+        let encoder = TiffEncoder::new_with_compression(
+            fs::File::create(&out_path).unwrap(),
+            CompressionType::Deflate(TiffDeflateLevel::Balanced),
```

Comment:
How long does this test take? I could imagine that compressing all the input files with balanced compression could take a while, though I don't know how many / how large they are.